File: backends/ruby-hanami/app/infrastructure/model/staff.rb
# frozen_string_literal: true
#
# スタッフモデルを定義するモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Infrastructure
  module Model
    # スタッフの永続化モデルです。
    # @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
    Staff = Struct.new(
      :id, :name, :email, :provider, :provider_id, :avatar, :role, :last_login_at,
      :created_at, :created_by, :updated_at, :updated_by,
      :deleted_at, :deleted_by, :version,
      keyword_init: true
    )
  end
end

File: backends/go-echo/ent/schema/staff.go
package schema

import (
	"entgo.io/ent"
	"entgo.io/ent/dialect/entsql"
	"entgo.io/ent/schema"
	"entgo.io/ent/schema/field"
	"time"
)

// Staff holds the schema definition for the Staff entity.
type Staff struct {
	ent.Schema
}

func (Staff) Annotations() []schema.Annotation {
	return []schema.Annotation{
		entsql.Annotation{Table: "staffs"},
	}
}

func (Staff) Fields() []ent.Field {
	return []ent.Field{
		field.Uint("id").Positive().Immutable(),
		field.String("name"),
		field.String("email"),
		field.Int("provider"),
		field.String("provider_id"),
		field.String("avatar").Optional().Nillable(),
		field.Int("role").Default(2),
		field.Time("last_login_at").Optional().Nillable(),
		field.Time("created_at").Default(time.Now),
		field.Uint("created_by").Optional().Nillable(),
		field.Time("updated_at").Default(time.Now),
		field.Uint("updated_by").Optional().Nillable(),
		field.Time("deleted_at").Optional().Nillable(),
		field.Uint("deleted_by").Optional().Nillable(),
		field.Int("version").Default(0),
	}
}

func (Staff) Edges() []ent.Edge {
	return nil
}
